unstructured_ingest/v2/processes/connectors/astradb.py

```diff
@@ -45,6 +89,154 @@ class AstraDBUploadStagerConfig(UploadStagerConfig):
+        "Note that the collection name must only include letters, "
+        "numbers, and underscores."
+    )
+    keyspace: Optional[str] = Field(default=None, description="The Astra DB connection keyspace.")
+    namespace: Optional[str] = Field(
+        default=None,
+        description="The Astra DB connection namespace.",
+        deprecated="Please use 'keyspace' instead.",
+    )
+
+
+class AstraDBDownloaderConfig(DownloaderConfig):
+    collection_name: str = Field(
+        description="The name of the Astra DB collection. "
+        "Note that the collection name must only include letters, "
+        "numbers, and underscores."
+    )
+    keyspace: Optional[str] = Field(default=None, description="The Astra DB connection keyspace.")
+    namespace: Optional[str] = Field(
+        default=None,
+        description="The Astra DB connection namespace.",
+        deprecated="Please use 'keyspace' instead.",
+    )
+
+
+class AstraDBUploaderConfig(UploaderConfig):
+    collection_name: str = Field(
+        description="The name of the Astra DB collection. "
+        "Note that the collection name must only include letters, "
+        "numbers, and underscores."
+    )
+    embedding_dimension: int = Field(
+        default=384, description="The dimensionality of the embeddings"
+    )
+    keyspace: Optional[str] = Field(default=None, description="The Astra DB connection keyspace.")
+    namespace: Optional[str] = Field(
+        default=None,
+        description="The Astra DB connection namespace.",
+        deprecated="Please use 'keyspace' instead.",
+    )
+    requested_indexing_policy: Optional[dict[str, Any]] = Field(
+        default=None,
+        description="The indexing policy to use for the collection.",
+        examples=['{"deny": ["metadata"]}'],
+    )
+    batch_size: int = Field(default=20, description="Number of records per batch")
+
+
+@dataclass
+class AstraDBIndexer(Indexer):
+    connection_config: AstraDBConnectionConfig
+    index_config: AstraDBIndexerConfig
+
+    def precheck(self) -> None:
+        try:
+            get_astra_collection(connection_config=self.connection_config, config=self.index_config)
+        except Exception as e:
+            logger.error(f"Failed to validate connection {e}", exc_info=True)
+            raise SourceConnectionError(f"failed to validate connection: {e}")
+
+    @requires_dependencies(["astrapy"], extras="astradb")
+    def get_collection(self) -> "AstraDBCollection":
+        return get_astra_collection(
+            connection_config=self.connection_config,
+            config=self.index_config,
+        )
+
+    @requires_dependencies(["astrapy"], extras="astradb")
+    def astra_record_to_file_data(self, astra_record: Dict[str, Any]) -> FileData:
+        return FileData(
+            identifier=astra_record["_id"],
+            connector_type=CONNECTOR_TYPE,
+            source_identifiers=SourceIdentifiers(
+                fullpath=astra_record["_id"] + ".txt",
```

Comment:
@roman Should we add the .txt here or down in the Downloader?

Reply:
Given that astradb is a database, we probably want to be using the batch support here, meaning we index by pulling in the ids and expanding that in the downloader. Take a look at the elasticsearch source implementation [here](https://github.com/Unstructured-IO/unstructured-ingest/blob/main/unstructured_ingest/v2/processes/connectors/elasticsearch.py#L173)